### PR TITLE
(FACT-2135) Create ssh resolver spec.

### PR DESCRIPTION
**Pull request by gigicampean**

### CODE REVIEW

File: spec/facter/resolvers/ssh_resolver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        file_address = directory_path + file_name
+        allow(File).to receive(:directory?).with(directory_path).and_return(true)
+        allow(File).to receive(:file?).with(file_address).and_return(true)
+        allow(File).to receive(:read).with(file_address).and_return(file_content)
+      end
+    end
+  end
+  context '#resolve ssh facts' do
+    let(:file_content) { [] }
+    it 'detects ssh dir,file,fact' do
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/ssh_resolver_spec.rb]: This test isn't really helpful.
It just verifies that code works if the file is empty, and in your case all the files are empty.

You should add tests that with some content.

File: spec/facter/facts/fedora/ssh_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    it 'returns a fact' do
+      result = []
+      ssh = Facter::Ssh.new(Facter::FingerPrint.new('test', 'test'), 'ecdsa', 'test', 'ecdsa')
+      result << ssh
+      result_fact = { ssh.name.to_sym =>
+                          { 'fingerprints'.to_sym =>
+                                { 'sha1'.to_sym => ssh.fingerprint.sha1,
+                                  'sha256'.to_sym => ssh.fingerprint.sha256 },
+                            'key'.to_sym => ssh.key,
+                            'type'.to_sym => ssh.type } }
```
Comment (oanatmaria) [inline spec/facter/facts/fedora/ssh_spec.rb]: :fingerprints .  :sha1    :key .   you can make them symbol directly

File: spec/fixtures/ecdsa [hunk 3]
```diff
@@ -0,0 +1 @@
+ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBNG0AgjrPXt5/osLsmECV/qISOwaRmDW1yNSHZiAJvZ6p6ZUilg5vqtSskaUsT5XN8J2amVktN6wOHDwtWiEbpM=
```
Comment (oanatmaria) [inline spec/fixtures/ecdsa]: enter

File: spec/fixtures/ecdsa_key [hunk 4]
```diff
@@ -0,0 +1 @@
+AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBNG0AgjrPXt5/osLsmECV/qISOwaRmDW1yNSHZiAJvZ6p6ZUilg5vqtSskaUsT5XN8J2amVktN6wOHDwtWiEbpM=
```
Comment (oanatmaria) [inline spec/fixtures/ecdsa_key]: enter

File: spec/facter/resolvers/ssh_resolver_spec.rb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                 key: load_fixture('rsa_key').read, name: 'rsa')
+        end
+
+        it 'returns fact' do
+          expect(Facter::Resolvers::SshResolver.resolve(:ssh)).to eq([rsa_result, result])
+        end
+      end
+      # context 'ecdsa file doesnt exist' do
+      #   it { allow(File).to receive(:file?).with('ssh_host_ecdsa_key.pub').and_return(false) }
+      # end
```
Comment (oanatmaria) [inline spec/facter/resolvers/ssh_resolver_spec.rb]: remove commented code

**Review by BogdanIrimie — APPROVED** [review]

**Pull request merged**